components/pkg-export-docker/src/docker.rs

```diff
@@ -152,21 +152,26 @@ impl<'a> DockerImage {
     /// * If a registry login is not successful
     /// * If a pushing one or more of the image tags fails
     /// * If a registry logout is not successful
-    pub fn push(&self, ui: &mut UI, credentials: &Credentials) -> Result<()> {
+    pub fn push(
+        &self,
+        ui: &mut UI,
+        credentials: &Credentials,
+        registry_url: &Option<&str>,
```

Comment:
It's _super_ rare to see `&Option`.  I think you just want `Option` here, because `registry_url` can either be `None` or `Some("some_url")` right?